client/me/security-account-email/index.jsx

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import AccountSettingsEmailAddress from 'me/account/email-address';
+import { Card } from '@automattic/components';
+import DocumentHead from 'components/data/document-head';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import FormButton from 'components/forms/form-button';
+import HeaderCake from 'components/header-cake';
+import isFetchingUserSettings from 'state/selectors/is-fetching-user-settings';
+import isPendingEmailChange from 'state/selectors/is-pending-email-change';
+import Main from 'components/main';
+import MeSidebarNavigation from 'me/sidebar-navigation';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+import QueryUserSettings from 'components/data/query-user-settings';
+import ReauthRequired from 'me/reauth-required';
+import { saveUserSettings } from 'state/user-settings/actions';
+import twoStepAuthorization from 'lib/two-step-authorization';
+
+class SecurityAccountEmail extends React.Component {
+	static propTypes = {
+		path: PropTypes.string,
+		translate: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			emailIsInvalid: true,
+		};
+		this.isSubmitting = this.isSubmitting.bind( this );
+		this.setEmailValidationState = this.setEmailValidationState.bind( this );
+		this.submitEmailForm = this.submitEmailForm.bind( this );
+	}
+
+	setEmailValidationState( emailIsInvalid ) {
+		this.setState( { emailIsInvalid } );
+	}
+
+	isSubmitting() {
+		return this.props.isFetchingUserSettings;
+	}
+
+	canSubmit() {
+		return ! this.state.emailIsInvalid && ! this.props.isPendingEmailChange;
+	}
+
+	submitEmailForm( event ) {
+		event.preventDefault();
+
+		this.setState( { isSubmitting: true } );
+
+		this.props.saveUserSettings( null, function ( error ) {
+			const errorMessage = error.message
+				? error.message
+				: this.props.translate( 'There was a problem updating your email address.' );
+			this.props.errorNotice( errorMessage );
```

Comment:
When I've simulated an error when updating the email (added a hardcoded `return new WP_Error` onn the backend), this line threw an error, because `this.props` was undefined. I'm guessing that's because the callback is a regular function, not the arrow type (`( error ) => {}`) that automatically binds to `this` as you'd expect. JavaScript ❤️

Reply:
Not quite copy-pasta, but close enough. I worked around by using destructured variables along with a closure, as well as simplifying the code path.